aspnetcore/security/anti-request-forgery.md

```diff
@@ -207,65 +165,20 @@ For more information, see <xref:Microsoft.AspNetCore.Builder.CookieAuthenticatio
-                new CookieOptions() { HttpOnly = false });
-        }
-
-        return next(context);
-    });
-}
-```
+The preceding example sets a cookie named `XSRF-TOKEN`. The client can read this cookie and provide its value as a header attached to AJAX requests. For example, Angular includes [built-in XSRF protection](https://angular.io/guide/http#security-xsrf-protection) that reads a cookie named `XSRF-TOKEN` by default.
 
 ### Require antiforgery validation
 
-[ValidateAntiForgeryToken](xref:Microsoft.AspNetCore.Mvc.ValidateAntiForgeryTokenAttribute) is an action filter that can be applied to an individual action, a controller, or globally. Requests made to actions that have this filter applied are blocked unless the request includes a valid antiforgery token.
-
-```csharp
-[HttpPost]
-[ValidateAntiForgeryToken]
-public async Task<IActionResult> RemoveLogin(RemoveLoginViewModel account)
-{
-    ManageMessageId? message = ManageMessageId.Error;
-    var user = await GetCurrentUserAsync();
-
-    if (user != null)
-    {
-        var result = 
-            await _userManager.RemoveLoginAsync(
-                user, account.LoginProvider, account.ProviderKey);
-
-        if (result.Succeeded)
-        {
-            await _signInManager.SignInAsync(user, isPersistent: false);
-            message = ManageMessageId.RemoveLoginSuccess;
-        }
-    }
+The [ValidateAntiForgeryToken](xref:Microsoft.AspNetCore.Mvc.ValidateAntiForgeryTokenAttribute) action filter can be applied to an individual action, a controller, or globally. Requests made to actions that have this filter applied are blocked unless the request includes a valid antiforgery token:
 
-    return RedirectToAction(nameof(ManageLogins), new { Message = message });
-}
-```
+:::code language="csharp" source="anti-request-forgery/samples/6.x/AntiRequestForgerySample/Snippets/Controllers/HomeController.cs" id="snippet_ValidateAntiForgeryToken" highlight="2":::
 
 The `ValidateAntiForgeryToken` attribute requires a token for requests to the action methods it marks, including HTTP GET requests. If the `ValidateAntiForgeryToken` attribute is applied across the app's controllers, it can be overridden with the `IgnoreAntiforgeryToken` attribute.
 
-> [!NOTE]
-> ASP.NET Core doesn't support adding antiforgery tokens to GET requests automatically.
-
 ### Automatically validate antiforgery tokens for unsafe HTTP methods only
 
 ASP.NET Core apps don't generate antiforgery tokens for safe HTTP methods (GET, HEAD, OPTIONS, and TRACE). Instead of broadly applying the `ValidateAntiForgeryToken` attribute and then overriding it with `IgnoreAntiforgeryToken` attributes, the [AutoValidateAntiforgeryToken](xref:Microsoft.AspNetCore.Mvc.AutoValidateAntiforgeryTokenAttribute) attribute can be used. This attribute works identically to the `ValidateAntiForgeryToken` attribute, except that it doesn't require tokens for requests made using the following HTTP methods:
```

Comment:
Should we change safe to [Idempotent](https://developer.mozilla.org/en-US/docs/Glossary/Idempotent)? BTW, I like monzilla links. Use Wikipedia only as last resort.

Reply:
I don't think we can use idempotent here, because DELETE is classed as idempotent but it's unsafe.

Reply:
for ***safe*** HTTP methods (GET, HEAD, OPTIONS, and TRACE).

Reply:
I revisited this and removed the line in question. We list those same HTTP methods at the end of this paragraph, so I think it's redundant to list them here. I think it's a little out of context too, because this section talks about enabling auto-validation but the line talks about auto generation. I think we already cover the generation side in [ASP.NET Core antiforgery configuration](https://review.docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?view=aspnetcore-6.0&branch=pr-en-us-24202#aspnet-core-antiforgery-configuration):

> The automatic generation of antiforgery tokens for HTML form elements happens when the <form> tag contains the method="post" attribute and...

Does that sound sensible?